models/lpjguess/R/model2netcdf.LPJGUESS.R

```diff
@@ -0,0 +1,131 @@
+  
+  # mnee "monthly net ecosystem C exchange" in kgC/m2/month to NEE kgC/m2/s
+  if("mnee.out" %in% lpjguess.out.files) nee=lpjguess.output[[which(lpjguess.out.files=="mnee.out")]][,4:15]/m.to.s
+  
+  # mlai "monthly Leaf Area Index" in m2/m2 to LAI m2/m2
+  if("mnee.out" %in% lpjguess.out.files) lai=lpjguess.output[[which(lpjguess.out.files=="mlai.out")]][,4:15]
+  
+  
+  
+  ### Loop over years in LPJ-GUESS output to create separate netCDF outputs
+  for (y in years){
+    
+    print(paste("---- Processing year: ", y))
+    
+    ## Setup outputs for netCDF file in appropriate units
+    
+    ## TODO: generalize for all possible outputs, both yearly and monthly
+    
+    output <- list()
+
+    output[[1]] = gpp[which(years==y),] # GPP in kgC/m2/s
+    output[[2]] = npp[which(years==y),] # NPP in kgC/m2/s
+    output[[3]] = arp[which(years==y),] # AutoResp in kgC/m2/s
+    output[[4]] = hrp[which(years==y),] # HeteroResp in kgC/m2/s
+    output[[5]] = nee[which(years==y),] # NEE in kgC/m2/s
+    output[[6]] = lai[which(years==y),] # LAI in m2/m2
+
+
+    
+    
+    #******************** Declare netCDF dimensions and variables ********************#
+    t <- ncdim_def(name = "time",
+                   units = paste0("days since ", y, "-01-01 00:00:00"),
+                   vals = 1:12,
+                   calendar = "standard", unlim = TRUE)
+    lat <- ncdim_def("lat", "degrees_east",
+                     vals =  as.numeric(sitelat),
+                     longname = "station_latitude") 
+    lon <- ncdim_def("lon", "degrees_north",
+                     vals = as.numeric(sitelon),
+                     longname = "station_longitude")
+    
+    
+    mstmipvar <- PEcAn.utils::mstmipvar
+    
+    var <- list()
+
+    var[[1]]  <- mstmipvar("GPP", lat, lon, t, NA)
+    var[[2]]  <- mstmipvar("NPP", lat, lon, t, NA)
+    var[[3]]  <- mstmipvar("AutoResp", lat, lon, t, NA)
+    var[[4]]  <- mstmipvar("HeteroResp", lat, lon, t, NA)
+    var[[5]]  <- mstmipvar("NEE", lat, lon, t, NA)
+    var[[6]]  <- mstmipvar("LAI", lat, lon, t, NA)
+    
+    #******************** Declare netCDF variables ********************#
+    
+    
+    ### Output netCDF data
+    nc <- nc_create(file.path(outdir, paste(y,"nc", sep=".")), var)
+    varfile <- file(file.path(outdir, paste(y, "nc", "var", sep=".")), "w")
+    for(i in 1:length(var)){
+      #print(i)
+      ncvar_put(nc,var[[i]],output[[i]])  
+      cat(paste(var[[i]]$name, var[[i]]$longname), file=varfile, sep="\n")
+    }
+    close(varfile)
+    nc_close(nc)
+    
+  } ### End of year loop
+  
+  
+} ### End of function
+#==================================================================================================#
+
+
+####################################################################################################
+### EOF.  End of R script file.              
+####################################################################################################
```

Comment:
actually this is a subset of LPJ-GUESS outputs, I only went with the ones I was sure that they are an exact match with MsTMIP variables. Also these are only the monthly ones, there are annual outputs too. I was planning to generalize everything in the end, would that be OK?